sherpa/cpp_api/websocket/online-websocket-client.cc

```diff
@@ -132,15 +134,19 @@ class Client {
     }
   }
 
-  void SendMessage(connection_hdl hdl) {
+  void SendMessage(connection_hdl hdl, std::chrono::time_point<std::chrono::steady_clock> start_time) {
     int32_t num_samples = samples_.numel();
     int32_t num_messages = num_samples / samples_per_message_;
 
     websocketpp::lib::error_code ec;
-
+    auto time = std::chrono::steady_clock::now();
+    int elapsed_time_ms = std::chrono::duration_cast<std::chrono::milliseconds>(time-start_time).count();
+    if (elapsed_time_ms < int(seconds_per_message_*num_sent_messages_*1000)) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(int(seconds_per_message_*num_sent_messages_*1000-elapsed_time_ms)));
```

Comment:
For async programming, I think it is better to start a timer to invoke a function at a given time point.  `sleep` will block the current thread.
(Not sure if you are familiar with usages of timer and its callback in `asio`. If not, I can help with it.)